climate-si7020/src/lib.rs

```diff
@@ -71,15 +76,17 @@ impl<'a> Climate<'a> {
         Ok(())
     }
 
-    pub fn read_temperature(&mut self) -> io::Result<f64> {
+    pub fn read_temperature(&mut self, unit: TemperatureUnit) -> io::Result<f64> {
         let mut buf = [0; 2];
         try!(self.read(&[Command::TempHold], &mut buf));
 
         let raw_temp = ((buf[0] as u16) << 8) + (buf[1] as u16);
         let mut temp = ((raw_temp as f64) * TEMPERATURE_SLOPE) - TEMPERATURE_OFFSET;
 
-        // Convert to fahrenheit.
-        temp = (temp * (9.0/5.0)) + 32.0;
+        match unit {
+            TemperatureUnit::Celsius => (),
+            TemperatureUnit::Fahrenheit => temp = temp * (9.0/5.0) + 32.0,
+        }
 
         Ok(temp)
```

Comment:
If you move this to inside the match on `unit` you can drop the `mut` part from `let mut temp =`

```rust
pub fn read_temperature(&mut self, unit: TemperatureUnit) -> io::Result<f64> {
    // ...

    match unit {
        TemperatureUnit::Celsius => Ok(temp),
        TemperatureUnit::Fahrenheit => Ok(temp * (9.0/5.0) + 32.0),
    }
}
```